docs/auth.md

```diff
@@ -0,0 +1,96 @@
+
+- Password strength checking
+- Throttling of login attempts
+- Authentication against third-parties (OAuth, for example)
+- Object-level permissions
+
+#### Users
+
+User objects are the core of the authentication system. They typically represent the people interacting with your site and are used to enable things like restricting access, registering user profiles, associating content with creators, etc. Only one class of user exists in Soul's authentication framework, i.e., 'superusers' or types of users are just user objects with special attributes set, not different classes of user objects.
+
+The primary attributes of the default user are:
+
+- username
+- password
+- email
+- first_name
+- last_name
+
+Note that when Soul boots up, it looks for a table called "\_users" which holds the User objects mentioned above.
+
+Creating superusers:
+
+Create superusers using the createsuperuser command:
+
+```
+$ node src/server.js createsuperuser --username=joe --password=strongstring
+```
+
+#### Permissions
+
+Soul comes with a built-in permissions system. It provides a way to assign permissions to specific users and groups of users.
+
+Soul uses permissions as follows:
+
+- Access to view objects is limited to users with the "read" permission for that type of object.
+- Access to add an object is limited to users with the "create" permission for that type of object.
+- Access to change an object is limited to users with the "update" permission for that type of object.
+- Access to delete an object is limited to users with the "delete" permission for that type of object.
+
+Default Permissions:
+
+By default, Soul ensures that four default permissions – create, update, delete, and read – are created for each table defined in one of your databases.
+
+| table_name | can_create | can_read | can_update | can_delete |
+| ---------- | ---------- | -------- | ---------- | ---------- |
+| albums     | false      | true     | true       | false      |
+
+In this scenario, calling POST (create) and GET (read) requests to /tables/albums/rows is allowed by any user (even anonymous), while PUT (update) and DELETE (delete, of course) requests are not allowed.
+
+When Soul boots up, it checks for the existence of a table called \_default_permissions.
+
+#### Roles
+
+Roles are a generic way of categorizing users so you can apply permissions or some other label to those users. A user can belong to any number of roles.
+
+A user with a role automatically has the permissions granted to that role. For example, if the role "editor" has the permission can_update_posts, any user with that role will have that permission.
+
+Beyond permissions, roles are a convenient way to categorize users and give them some label or extended functionality.
+
+#### Authentication
+
+Soul uses cookies and middleware to hook the authentication system into request objects.
+
```

Comment:
You're right, I intentionally used the term "cookie" because it is considered a best practice to send refresh tokens over HTTP-only cookies instead of including them in the response body. Since authentication is essentially a relationship between the server and the end user, using a cookie for this purpose is a logical choice.